trade-finance-manager-ui/server/controllers/case/index.js

```diff
@@ -45,13 +44,11 @@ const getCaseDeal = async (req, res) => {
 
   const { submissionType } = deal.dealSnapshot;
 
-  const dealCancellationIsEnabled = isDealCancellationEnabled(submissionType, user);
-  let hasDraftCancellation = false;
+  const dealCancellation = await api.getDealCancellation(dealId, userToken);
 
-  if (dealCancellationIsEnabled) {
-    const cancellation = await api.getDealCancellation(dealId, userToken);
-    hasDraftCancellation = !isEmpty(cancellation);
-  }
+  const dealCancellationPermitted = canDealBeCancelled(submissionType, user, dealCancellation.status);
+
+  const hasDraftCancellation = dealCancellationPermitted && dealCancellation?.status === TFM_DEAL_CANCELLATION_STATUS.DRAFT;
```

Comment:
Maybe could move into a helper directory, like `canDealBeCancelled`?

Reply:
This logic will only be used here, so keen to keep it as is if that's ok!